Fix(#580): Rename sheet_name to sheet at to_excel

Avasse opened the pull request:
Reopen of : https://github.com/liam2/larray/pull/583 (Sorry don't know why i couldn't reopen it...)

I renamed every "sheet_name" to "sheet" in to_excel def.
I added a changelog entry in doc/source/changes/version_0_28.rst.inc section "Miscellaneous improvements"

Also, i accept my contribution and all future contributions to be released under the terms of the GPLv3 license.

Avasse commented:
@gdementen Thanks for your help, i finally couldn't figure with the deprecate_kwarg decorator... Could you help me a bit on this ? I searched for it but i don't know how to use it. 

Thanks for your time

alixdamman commented:
@Avasse add `@deprecate_kwarg('sheet_name', 'sheet')` on top of `to_excel`.

Avasse commented:
@alixdamman  Did it, i hope it was what you expected me to do, sorry i'm not a python developer i'm quite lost but i try my best :)

gdementen commented on larray/core/array.py:
```diff
@@ -6024,14 +6028,14 @@ def to_excel(self, filepath=None, sheet_name=None, position='A1', overwrite_file
 
             if new_workbook:
                 sheet = wb.sheets[0]
```
You also need to rename this variable, because this used to be a different variable and now it has the same name than the argument, which will break the code below.

gdementen commented on larray/core/array.py:
```diff
@@ -5994,7 +5995,10 @@ def to_excel(self, filepath=None, sheet_name=None, position='A1', overwrite_file
         """
-        sheet_name = _translate_sheet_name(sheet_name)
+        
+        @deprecate_kwarg('sheet_name', 'sheet')
```
this line/decorator needs to be just above the "def to_excel(" line.

alixdamman commented:
@Avasse I should have been more explicit. `deprecate_kwarg` is a decorator (see https://docs.python.org/3/whatsnew/2.4.html#pep-318-decorators-for-functions-and-methods). You must use it as follow:

```python
@deprecate_kwarg('sheet_name', 'sheet')
def to_excel(....)
```

Avasse commented:
@gdementen I followed your request, i think it's pretty good now (I thought it was good since 3 answers, i hope this time all is fine 💪 )

gdementen commented:
It is not 😢. One thing we should have pointed out earlier (you were not supposed to know that) is that many Excel-related tests are not run on Travis CI, so you need to run them on your machine using pytest.
For example, go to larray/tests and run "pytest ."

gdementen commented on larray/core/array.py:
```diff
@@ -6023,15 +6028,15 @@ def to_excel(self, filepath=None, sheet_name=None, position='A1', overwrite_file
-            elif sheet_name is not None and sheet_name in wb:
-                sheet = wb.sheets[sheet_name]
+                first_sheet = wb.sheets[0]
+                if sheet is not None:
```
Maybe our tests are not good enough to catch this, but this is buggy... What you need to do is rename all occurrences of "sheet" (before the sheet_name -> sheet rename, obviously) to something else. E.g. sheetobj

Avasse commented:
@gdementen I did it, i don't know if it works cause i got errors on make_realease import. Actually i came from scratch and renamed all `sheet` occurences for `sheetobj`, then i renamed all `sheet_name` occurences with `sheet`

gdementen commented on doc/source/changes/version_0_28.rst.inc:
```diff
@@ -283,6 +283,9 @@ Miscellaneous improvements
   Closes :issue:`549`.
 
+* Renamed argument `sheet_name` to `sheet` on array.py `to_excel` def.
+  Rename `sheet` initial occurences to `sheetobj` to prevent from rename errors.
```
Please remove this line. The changelog is for users so they do not care about such details.

gdementen commented on doc/source/changes/version_0_28.rst.inc:
```diff
@@ -283,6 +283,9 @@ Miscellaneous improvements
 
   Closes :issue:`549`.
 
+* Renamed argument `sheet_name` to `sheet` on array.py `to_excel` def.
```
you should not mention array.py since the changelog is for users and they do not care where the method is.
You might want to write something like this instead:
> Renamed  `sheet_name` of LArray.to_excel to `sheet` since it can also be an index (closes :issue:`580`).

Avasse commented:
@gdementen "The code changes look good this time." Best phrase i've ever read... 😃

gdementen commented:
Could you rebase your PR on top of (the lastest) master?

Avasse commented:
@gdementen I tried, don't know if i rebased correctly the PR .... 😢

gdementen commented:
while this page shows "This branch cannot be rebased due to conflicts", it means it isn't. You need to update your fork (search google for help how to do that) then rebase your branch on top of master

gdementen commented:
(sorry, closed by mistake)

Avasse commented:
@gdementen Hello, i tried to follow this guide : https://help.github.com/articles/syncing-a-fork/

And when i try to merge upstream/master i'm up to date, so i think it's good... I can't see the "This branch cannot be rebased due to conflicts" Could you tell me where i'm supposed to see it ? 

Have a good day 😄

Avasse commented:
@gdementen Hi, i get an error that i can't understand....
>       sheet = _translate_sheet_name(sheet)
E       UnboundLocalError: local variable 'sheet' referenced before assignment
At array.py line 6032. 

Fact is that i just renamed sheet_name to sheet at this line and i didn't had any error before mergin (while code didn't changed after all).

alixdamman commented:
@Avasse Just take a look in tab "Files Changed" of this page. You will see that you have also changed something in the function `read_excel` you shouldn't have.

Avasse commented:
@alixdamman @gdementen There we go ? 😄

alixdamman commented:
@Avasse github currently says "This branch cannot be rebased due to conflicts" (see below)

Avasse commented:
![img_20180221_131406](https://user-images.githubusercontent.com/8959855/36479481-26666ac2-1709-11e8-9348-a51a44ab0fba.png)
@alixdamman Don't understand how but I see "This branch has no conflicts with the base branch" (see screen) Thanks for your help on this PR again...

alixdamman commented:
@Avasse Not anymore but the remaining conflict is located in the changelog file and should easy to fix

![capture](https://user-images.githubusercontent.com/23628987/36490297-3db41c64-1728-11e8-8286-83992d49047e.PNG)

Avasse commented:
@alixdamman Resolved the easy conflict, hope this is fine on your side too ?

Avasse commented:
@alixdamman @gdementen Sorry for the inconvenience... Could i have any news about the state of this PR please ? Is it ok for you to merge it ?

alixdamman commented:
@Avasse It still in conflict. Currently, this project evolves quite rapidly... sorry.

To avoid extra 'merge' commits, you may use `git pull --rebase (...)` instead of `git pull (...)`:
- https://coderwall.com/p/7aymfa/please-oh-please-use-git-pull-rebase
- https://stackoverflow.com/questions/18930527/difference-between-git-pull-and-git-pull-rebase

Avasse commented:
@alixdamman I used git pull --rebase, actually i don't know how to know if there's conflicts since i always see the button "This branch has no conflicts with the base branch" And everything up to date with upstream/master so.... 

Could i have some news asap ? I needed to end this PR until tomorrow and everything seems always fine on my screen so i can't fix errors that doesn't appear....

alixdamman commented:
@gdementen What do you see? Do you see remaining conflicts?

Avasse commented:
Still not... 
![capture](https://user-images.githubusercontent.com/8959855/36556269-e6450618-1804-11e8-9b72-3972f2a12539.png)

gdementen commented:
Yes, still a conflict. I suppose this is due to the apparently duplicate commits. I guess squashing all these commits together into one via an interactive rebase would solve the problem.

gdementen commented:
@Avasse If you sign out of Github I **suppose**, you will see the conflict. This is really strange. Probably a bug on GitHub.

Avasse commented:
@gdementen Even if I sign out i can't ee the conflict, 1 week about conflicts lol, I should have just re-open a new PR after my changes were accepted, i'll open a new one tomorrow morning, that will probably avoid the conflict (now that i know what I have to change, it will be easy).
